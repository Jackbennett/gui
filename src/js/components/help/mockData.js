export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
