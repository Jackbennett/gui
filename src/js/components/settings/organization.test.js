import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import MyOrganization from './organization';
import { undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('MyOrganization Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      app: {
        features: {
          isHosted: false
        }
      },
      users: {
        organization: {
          id: 1,
          name: 'test',
          tenant_token: 'test'
        }
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyOrganization />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
