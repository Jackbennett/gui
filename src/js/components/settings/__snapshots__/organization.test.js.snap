// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyOrganization Component renders correctly 1`] = `
<div
  className="margin-top-small"
>
  <h2
    style={
      Object {
        "marginTop": "15px",
      }
    }
  >
    My organization
  </h2>
  <div>
    <ul
      className="MuiList-root MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-divider Mui-disabled"
        disabled={true}
        style={
          Object {
            "maxWidth": "500px",
          }
        }
      >
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Organization name
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            test
          </p>
        </div>
      </li>
      <div
        className="flexbox"
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "display": "inline-block",
              "width": "500px",
            }
          }
        >
          <li
            className="MuiListItem-root attributes MuiListItem-gutters Mui-disabled"
            disabled={true}
          >
            <div
              className="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                <div>
                  <h4
                    style={
                      Object {
                        "display": "inline",
                        "paddingRight": "10px",
                      }
                    }
                  >
                    Token
                  </h4>
                  <div
                    className="tooltip info"
                    data-event="click focus"
                    data-for="token-help"
                    data-tip={true}
                    id="token-info"
                    style={
                      Object {
                        "display": "inline",
                        "position": "relative",
                        "top": "6px",
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                      />
                    </svg>
                  </div>
                  <div
                    className="__react_component_tooltip place-top type-dark"
                    data-id="tooltip"
                    id="token-help"
                  >
                    <h3>
                      Tenant token
                    </h3>
                    <p
                      style={
                        Object {
                          "color": "#DECFD9",
                          "margin": "1em 0",
                        }
                      }
                    >
                      This token is unique for your organization and ensures that only devices that you own are able to connect to your account.
                    </p>
                    <p
                      style={
                        Object {
                          "color": "#DECFD9",
                          "margin": "1em 0",
                        }
                      }
                    >
                      Set this variable in 
                      <i>
                        local.conf
                      </i>
                       in order to make the device recognize the organization to which it belongs.
                    </p>
                  </div>
                </div>
              </span>
              <div
                className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                title="test"
              >
                <div>
                  <span
                    className="break-all inventory-text tenant-token-text "
                    style={Object {}}
                  >
                    test
                  </span>
                   
                </div>
              </div>
            </div>
          </li>
          <hr
            className="MuiDivider-root"
          />
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "margin": "0 15px 15px",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
                />
              </svg>
            </span>
            Copy to clipboard
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <div>
          <p
            style={
              Object {
                "marginBottom": "30px",
              }
            }
          />
        </div>
      </div>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-divider Mui-disabled"
        disabled={true}
        style={
          Object {
            "maxWidth": "500px",
          }
        }
      >
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Current plan
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            Mender Enterprise
          </p>
        </div>
      </li>
    </ul>
    <p
      className="margin-top margin-left-small margin-right-small"
    >
      <a
        href="mailto:support@mender.io?subject=test: Change plan&body=Organization%20ID%3A%201%0AOrganization%20name%3A%20test%0APlan%20name%3A%20Mender Enterprise%0A%0AI%20would%20like%20to%20make%20a%20change%20to%20my%20Mender%20plan."
        target="_blank"
      >
        Upgrade your plan
      </a>
       
      or compare product plans at
       
      <a
        href="https://mender.io/pricing"
        target="_blank"
      >
        mender.io/pricing
      </a>
      .
    </p>
    <p
      className="margin-left-small margin-right-small"
    >
      <a
        href="mailto:support@mender.io?subject=test: Update billing&body=Organization%20ID%3A%201%0AOrganization%20name%3A%20test%0APlan%20name%3A%20Mender Enterprise%0A%0AI%20would%20like%20to%20make%20a%20change%20to%20my%20billing%20details."
        target="_blank"
      >
        Request to update your billing details
      </a>
    </p>
    <p
      className="margin-left-small margin-right-small"
    >
      <a
        href="mailto:support@mender.io?subject=test: Cancel subscription&body=Organization%20ID%3A%201%0AOrganization%20name%3A%20test%0APlan%20name%3A%20Mender Enterprise%0A%0APlease%20cancel%20my%20subscription.%0AReason%20%5Boptional%5D%3A"
        target="_blank"
      >
        Cancel subscription
      </a>
    </p>
  </div>
</div>
`;
